services/ui-src/src/Routes.js

```diff
@@ -25,7 +25,7 @@ export default function Routes() {
     const localEndpoint = config.s3.LOCAL_ENDPOINT;
     let s3Upload = s3AmplifyUpload;
     let s3URLResolver = s3AmplifyGetURL;
-    if (localEndpoint !== '') {
+    if (localLogin && localEndpoint !== '') {
```

Comment:
I'd do `localEndpoint &&` here, so as not to cross the streams. This would require not setting localEndpoint in CI by deleting it from ui-src/.env